### PR TITLE
Lee/create visual3

### PR DESCRIPTION
**Pull request by Hyukjoo-Lee**
[I've referenced this link](https://www.figma.com/file/TNqsfSN2F6twHtuodJCFsq/Inviteyou-(collaborate)?type=design&node-id=31-1540&mode=design&t=kAbRvf8yfNe7Unom-0)

- Designed a new visual3 template.
- Button functions associated with the form will be implemented later.
- Adjusted menu styles to eliminate hovering effects that occurred when users hovered over menu options
color change will occurs only when users actively click on a menu option

### CODE REVIEW

**Comment by HanKyungSung** [comment]
The font size is not responsive. It works well on mobile view but not on the desktop view.

> Desktop view:
image

**Comment by HanKyungSung** [comment]
Also, the top of the visual component is not responsive on the desktop view as well.
- Date box size
- Font for the text and alignment.

image

File: invitation-frontend/src/component/Visual3.tsx [hunk 1]
```diff
@@ -1,5 +1,723 @@
...
+    return (
+      <div key={`side-${img}`}>
+        <img src={img} alt={`Slide $(item)`} />
+      </div>
+    );
+  });
+
+  return (
+    <>
+      <MantineProvider
```
Comment (HanKyungSung) [inline invitation-frontend/src/component/Visual3.tsx:178]: I am not sure if we need `MaintineProvider` here. If we want to use it, should we have the break points here as well?

File: invitation-frontend/src/component/Visual3.tsx [hunk 2]
```diff
@@ -1,5 +1,723 @@
...
+        }}
+      >
+        <main
+          className={`${classes.invitationWrap} ${classes.invitationVisual}`}
+        >
+          <Container
+            className={classes.newVisual}
+            fluid
+            px={0}
+            size={RESPONSIVE_MOBILE ? 550 : 450}
```
Comment (HanKyungSung) [inline invitation-frontend/src/component/Visual3.tsx:217]: This is related to the breakpoints in the `MantineProvider`. Should we make it responsive depending on the Mantines's breakpoint we define using provider? or should we use the constant value?

File: invitation-frontend/src/component/Visual3.tsx [hunk 3]
```diff
@@ -1,5 +1,723 @@
...
+              <p className={classes.weddingCharacter}>
+                {bride} & {groom}
+              </p>
+              <p>@ {location}</p>
+              <p>
+                {monthNum}.{day}.{year} / {time}
+              </p>
+            </div>
+          </Container>
+          {/* TODO: Need to adjust text inside image */}
```
Comment (HanKyungSung) [inline invitation-frontend/src/component/Visual3.tsx]: I see this TODO comment now. Do you think we can have those changes in this PR?

File: invitation-frontend/src/component/Visual3.tsx [hunk 4]
```diff
@@ -1,5 +1,723 @@
...
+                align="left"
+                pt={50}
+              >
+                Daniel Han & Anna Smith
+              </Text>
+            </div>
+          </Container>
+
+          <Container
+            style={{
```
Comment (HanKyungSung) [inline invitation-frontend/src/component/Visual3.tsx]: I think we can refactor this inline style.

File: invitation-frontend/src/component/Visual3.tsx [hunk 5]
```diff
@@ -1,5 +1,723 @@
...
+                    <span>30</span>
+                  </td>
+                  <td>
+                    <span>31</span>
+                  </td>
+                </tr>
+              </tbody>
+            </table>
+          </Container>
+          <Container size={RESPONSIVE_MOBILE ? 550 : 450} py={85}>
```
Comment (HanKyungSung) [inline invitation-frontend/src/component/Visual3.tsx]: I see we keep using the same-size props with the same value. Can we refactor them?

File: invitation-frontend/src/component/Visual3.tsx [hunk 6]
```diff
@@ -1,5 +1,723 @@
...
+                <div className={classes.contactListLeft}>
+                  <figure>
+                    <Image
+                      src={imgBride}
+                      alt="brideImg"
+                      width={80}
+                      height={80}
+                    />
+                  </figure>
+                  <ul style={{ marginLeft: 15 }}>
```
Comment (HanKyungSung) [inline invitation-frontend/src/component/Visual3.tsx]: We may want to move away from the inline-style

File: invitation-frontend/src/component/Visual3.tsx [hunk 7]
```diff
@@ -1,5 +1,723 @@
...
+                      <Text
+                        color="#888"
+                        style={{ textDecoration: 'underline' }}
+                      >
+                        Check Her Phone number
+                      </Text>
+                    </li>
+                  </ul>
+                </div>
+                <ul style={{ display: 'flex' }}>
```
Comment (HanKyungSung) [inline invitation-frontend/src/component/Visual3.tsx]: This is as well. I think we should create new styles for each section.
ex)`contact-info-section-styles`

**Comment by HanKyungSung** [comment]
I didn't comment on every line that has an inline style. Can we refactor them as we discussed before?

**Comment by Hyukjoo-Lee** [comment]
> The font size is not responsive. It works well on mobile view but not on the desktop view.
> 
> > Desktop view:
> > image

**Comment by Hyukjoo-Lee** [comment]
> Also, the top of the visual component is not responsive on the desktop view as well.
> 
> * Date box size
> * Font for the text and alignment.
> 
> image



> > The font size is not responsive. It works well on mobile view but not on the desktop view.
> > > Desktop view:
> > > image

File: invitation-frontend/src/component/Visual3.tsx (hunk 1 above)
Comment (Hyukjoo-Lee) [reply invitation-frontend/src/component/Visual3.tsx:178]: I referenced this component, ../invitation-frontend/src/component/SamHan.tsx and I realized that we utilized the MantineProvider to set specific global styles for the entire design of a template

However, if you are talking about responsive design here, setting global breakpoints would be a good idea.

**Comment by Hyukjoo-Lee** [comment]
> > The font size is not responsive. It works well on mobile view but not on the desktop view.
> > > Desktop view:
> > > image

- The current code is only checking whether the viewport width is at most 767 pixels defining a var 'const RESPONSIVE_MOBILE = useMediaQuery('(max-width: 767px)');'
Do you think we need to add extra media queries to deal with this problem?
e.g.,
```javascript
const RESPONSIVE_WIDTH = {
  mobile: '767px',
  tablet: '768px',  
  desktop: '1024px'
};
```

File: invitation-frontend/src/component/Visual3.tsx (hunk 2 above)
Comment (Hyukjoo-Lee) [reply invitation-frontend/src/component/Visual3.tsx:217]: I think there could be pros/cons

if we set Mantine's breakpoints, we can implement a consistent responsive design across all components.
But, they may not align perfectly with the specific design requirements.

File: invitation-frontend/src/component/Visual3.tsx (hunk 3 above)
Comment (Hyukjoo-Lee) [reply invitation-frontend/src/component/Visual3.tsx]: As we discussed earlier, if arranging texts is difficult, we can place text in the middle. 
However, the issue was easily resolved with some style adjustments. 

I forgot to delete the comment before this PR. I'll delete it.

**Comment by Hyukjoo-Lee** [comment]
### Description

- Modified and enhanced the visual component to make it responsive on desktop view.
- Adjusted the positioning and sizing of various elements for better looking.
- Refactored inline styling as we discussed during the previous meeting.
- Encapsulate the duplicated responsive properties to reduce redundancy.

**Review by HanKyungSung — APPROVED** [review]

**Pull request merged**